merger/merger.go

```diff
@@ -96,7 +97,7 @@ const (
 // If an attribute is marked with a "# keep" comment, it will not be merged.
 // If a rule is marked with a "# keep" comment, the whole rule will not
 // be modified.
-func MergeFile(oldFile *rule.File, emptyRules, genRules []*rule.Rule, phase Phase, kinds map[string]rule.KindInfo) {
+func MergeFile(oldFile *rule.File, emptyRules, genRules []*rule.Rule, phase Phase, kinds map[string]rule.KindInfo) error {
```

Comment:
This would be an incompatible change to an exported function. Let's avoid that if we can.

Perhaps instead, `rule.LoadData`, `rule.LoadWorkspaceData`, and `rule.LoadMacroData` can call some common function, `func checkFile(f *File) error` that returns an error if there are multiple rules with the same name (we could conceivably check more invariants later).

I think this makes more sense than checking for duplicates in `MergeFile` anyway; arguably this is a pre-condition that `MergeFile` shouldn't have to worry about.